### PR TITLE
doc: add runtime reconfiguration design doc

### PR DESCRIPTION
**Pull request by xiang90**
Fix https://github.com/coreos/etcd/issues/3014

### CODE REVIEW

File: Documentation/runtime-reconf-design.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+Runtime reconfiguration is one of the hardest and most error prone features in a distributed system, especially in a consensus based system like etcd.
+
+Read on to learn about the design of etcd's runtime reconfiguration commands and how we tackled these problems.
+
+### Two Phase Config Changes Keep you Safe
+
+In etcd, every runtime reconfiguration has to go through [two phases](Documentation/runtime-configuration.md#add-a-new-member) for safety reasons. For example, to add a member you need to first inform cluster of new configuration and then start the new member.
+
+Phase 1 - Inform cluster of new configuration
```
Comment (yichengq) [inline Documentation/runtime-reconf-design.md]: empty line below needed

Comment (xiang90) [inline Documentation/runtime-reconf-design.md]: fixed

File: Documentation/runtime-reconf-design.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+### Permanent Loss of Quorum Requires New Cluster
+
+If a cluster permanently loses a majority of its members, a new cluster will need to be started from an old data directory to recover the previous state.
+
+It is entirely possible to force removing the failed members from the existing cluster to recover. However, we decided not to support this method since it bypasses the normal consensus committing phase, which is unsafe. If the member to remove is not actually dead or you force to remove different members through different members in the same cluster, you will end up with diverged cluster with same clusterID. This is very dangerous and hard to debug/fix afterwards. 
+
+If you have a correct deployment, the possibility of permanent majority lose is very low. But it is a severe enough problem that worth special care. However, we strongly suggest you to read the disaster recovery documentation and prepare for permanent majority lose before you put etcd into production.
```
Comment (yichengq) [inline Documentation/runtime-reconf-design.md]: remove However?

could you add the link for disaster doc here?

**Comment by yichengq** [comment]
LGTM. defer to @robszumski

**Comment by robszumski** [comment]
Don't we need to link to this new doc from the runtime reconfig doc?

**Comment by xiang90** [comment]
>  Don't we need to link to this new doc from the runtime reconfig doc?

We do link it from runtime reconfig doc. See the first changed file in the commits.

**Comment by robszumski** [comment]
LGTM

**Pull request merged**